### PR TITLE
`on_error` settings for `grape`

### PR DESCRIPTION
**Pull request by TonyCTHsu**
**2.0 Upgrade Guide notes**

🚨 Breaking change for Grape instrumentation: 

Replace `error_statuses` with `error_status_codes`. Change your configuration from string to array of ranges or integers. 

For example: 

Before:
```ruby
Datadog.configure do |c|
  c.tracing.instrument :grape, error_statuses: '400,500-599'
end
```

After: 
```ruby
Datadog.configure do |c|
  c.tracing.instrument :grape, error_status_codes: [400, 500..599]
end
```

**What does this PR do?**

1. Add `on_error` settings for `grape` instrumentation
2. Replace `error_statuses` with `error_status_codes` settings for `grape` instrumentation

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -792,8 +792,8 @@ connection.get('/foo')
...
-| `on_error` | Custom error handler invoked when a request raises an error. Provided `span` and `error` as arguments. Sets error on the span by deault. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
-| `error_status_codes`  | `DD_TRACE_FARADAY_ERROR_STATUS_CODES` | Range or Array of HTTP status codes that should be traced as errors.| `400...600` |
+| `on_error` | Custom error handler invoked when a request raises an error. Provided `span` and `error` as arguments. Sets error on the span by deault. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |**default: 400...600**
+| `error_status_codes`  | `DD_TRACE_FARADAY_ERROR_STATUS_CODES` | Defines HTTP status codes that would be traced as errors. Value could be a range(`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```
Comment (brett0000FF) [inline docs/GettingStarted.md:796]:
```suggestion
| `error_status_codes`  | `DD_TRACE_FARADAY_ERROR_STATUS_CODES` | Defines HTTP status codes that are traced as errors. Value can be a range (`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -824,7 +824,7 @@ end
 | Key              | Env Var                  | Description                                                                                                                   | Default |
 |------------------|--------------------------|-------------------------------------------------------------------------------------------------------------------------------|---------|
 | `enabled`        | `DD_TRACE_GRAPE_ENABLED` | Defines whether Grape should be traced. Useful for temporarily disabling tracing. `true` or `false`                           | `true`  |
-| `error_statuses` |                          | Defines a status code or range of status codes which should be marked as errors. `'404,405,500-599'` or `[404,405,'500-599']` | `nil`   |
+| `error_status_codes` | `DD_TRACE_GRAPE_ERROR_STATUS_CODES` | Defines HTTP status codes that would be traced as errors. Value could be a range(`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `500...600`   |
```
Comment (brett0000FF) [inline docs/GettingStarted.md:827]:
```suggestion
| `error_status_codes` | `DD_TRACE_GRAPE_ERROR_STATUS_CODES` | Defines HTTP status codes that are traced as errors. Value can be a range (`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `500...600`   |
```

File: docs/GettingStarted.md [hunk 3]
```diff
@@ -997,8 +997,7 @@ end
...
 | `distributed_tracing` |                                          | Enables [distributed tracing](#distributed-tracing)                                                                                                                                        | `true`      |
 | `split_by_domain`     |                                          | Uses the request domain as the service name when set to `true`.                                                                                                                            | `false`     |
-| `error_status_codes`  | `DD_TRACE_HTTPRB_ERROR_STATUS_CODES` | Range or Array of HTTP status codes that should be traced as errors.                                                                                                                       | `400...600` |
-
+| `error_status_codes`  | `DD_TRACE_HTTPRB_ERROR_STATUS_CODES` | Defines HTTP status codes that would be traced as errors. Value could be a range(`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```
Comment (brett0000FF) [inline docs/GettingStarted.md:1000]:
```suggestion
| `error_status_codes`  | `DD_TRACE_HTTPRB_ERROR_STATUS_CODES` | Defines HTTP status codes that are traced as errors. Value can be a range (`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```

**Review by marcotc — APPROVED** [review]

File: docs/GettingStarted.md [hunk 4]
```diff
@@ -1025,7 +1024,7 @@ end
...
 | `split_by_domain`     |                                          | Uses the request domain as the service name when set to `true`.                                                                                                                                | `false`      |
-| `error_status_codes`  | `DD_TRACE_HTTPCLIENT_ERROR_STATUS_CODES` | Range or Array of HTTP status codes that should be traced as errors.                                                                                                                           | `400...600`  |
+| `error_status_codes`  | `DD_TRACE_HTTPCLIENT_ERROR_STATUS_CODES` | Defines HTTP status codes that would be traced as errors. Value could be a range(`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600`  |
```
Comment (brett0000FF) [inline docs/GettingStarted.md:1027]:
```suggestion
| `error_status_codes`  | `DD_TRACE_HTTPCLIENT_ERROR_STATUS_CODES` | Defines HTTP status codes that are traced as errors. Value can be a range (`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600`  |
```

File: docs/GettingStarted.md [hunk 5]
```diff
@@ -1181,7 +1180,7 @@ content = Net::HTTP.get(URI('http://127.0.0.1/index.html'))
...
 | `distributed_tracing` |                                    | Enables [distributed tracing](#distributed-tracing)                                                                                                                                          | `true`      |
 | `split_by_domain`     |                                    | Uses the request domain as the service name when set to `true`.                                                                                                                              | `false`     |
-| `error_status_codes`  | `DD_TRACE_HTTP_ERROR_STATUS_CODES` | Range or Array of HTTP status codes that should be traced as errors.                                                                                                                         | `400...600` |
+| `error_status_codes`  | `DD_TRACE_HTTP_ERROR_STATUS_CODES` | Defines HTTP status codes that would be traced as errors. Value could be a range(`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```
Comment (brett0000FF) [inline docs/GettingStarted.md:1183]:
```suggestion
| `error_status_codes`  | `DD_TRACE_HTTP_ERROR_STATUS_CODES` | Defines HTTP status codes that are traced as errors. Value can be a range (`400...600`), or an array of ranges/integers `[403, 500...600]`. If configured with environment variable, use dash for range (`'400-599'`) and comma for adding element into an array (`'403,500-599'`) | `400...600` |
```

**Review by brett0000FF — APPROVED** [review]
Small style nits changing to present tense, avoiding [temporal words](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#tense), and fixing spacing before parenthetical. 👍

**Pull request merged**